### PR TITLE
add a direct link from files settings to app-token-settings

### PR DESCRIPTION
**Pull request by szaimen**
This is best reviewed like this: https://github.com/nextcloud/server/pull/37145/files?diff=unified&w=1

| Before | After |
|---|---|
| image | image |

<details>
<summary>For my own testing</summary>

```
docker run -it \
--name nextcloud-easy-test \
-p 8443:443 \
-e TRUSTED_DOMAIN=192.168.24.128 \
--volume="nextcloud_easy_test_npm_cache_volume:/var/www/.npm" \
-e SERVER_BRANCH=enh/noid/add-link-to-app-tokens \
-e COMPILE_SERVER=1 \
ghcr.io/szaimen/nextcloud-easy-test:latest
```

</details>

### CODE REVIEW

**Review by rakekniven — CHANGES_REQUESTED** [review]

File: apps/files/src/views/Settings.vue [hunk 1]
```diff
@@ -65,6 +65,12 @@
 					{{ t('files', 'Use this address to access your Files via WebDAV') }} ↗
 				</a>
 			</em>
+			<br />
+			<em>
+				<a :href="appPasswordUrl">
+					{{ t('files', 'If you have activated 2FA, you need to create and use a new app password by clicking here.') }}
```
[the remark this answers is not in the document]
Comment (szaimen) [reply apps/files/src/views/Settings.vue]: Fine by me, thanks!

File: apps/settings/src/components/AuthTokenSetupDialogue.vue [hunk 2]
```diff
@@ -21,18 +21,20 @@
...
-			:maxlength="120"
-			:disabled="loading"
-			:placeholder="t('settings', 'App name')"
-			@keydown.enter="submit">
-		<NcButton :disabled="loading || deviceName.length === 0"
-			type="primary"
-			@click="submit">
-			{{ t('settings', 'Create new app password') }}
-		</NcButton>
+		<a id="generate-app-token-section">
```
Comment (artonge) [inline apps/settings/src/components/AuthTokenSetupDialogue.vue:24]: Why not a div ?

**Review by artonge — CHANGES_REQUESTED** [review]

File: apps/settings/src/components/AuthTokenSetupDialogue.vue (hunk 2 above)
Comment (szaimen) [reply apps/settings/src/components/AuthTokenSetupDialogue.vue:24]: Linking to div with id="generate-app-token-section" didnt work in my testing...

**Review by rakekniven — APPROVED** [review]

File: apps/settings/src/components/AuthTokenSetupDialogue.vue (hunk 2 above)
Comment (szaimen) [reply apps/settings/src/components/AuthTokenSetupDialogue.vue:24]: I think it needs to be an anker...

**Review by artonge — APPROVED** [review]
I pushed a commit with:

- Target as div
- Underline decoration for links

**Comment by szaimen** [comment]
Thanks @artonge !

> * Target as div

Did you test if this really works?

**Comment by szaimen** [comment]
Just tested again and seems to work. I wonder why it didnt initially but okay.

**Comment by nickvergessen** [comment]
String freeze `¯\_(ツ)_/¯`

**Comment by szaimen** [comment]
27 only?

**Pull request merged**